cmd/podman/cp.go

```diff
@@ -0,0 +1,366 @@
+			// of the mount command.
+			return fmt.Errorf("cannot mount using driver %s in rootless mode", driver)
+		}
+
+		became, ret, err := rootless.BecomeRootInUserNS()
+		if err != nil {
+			return err
+		}
+		if became {
+			os.Exit(ret)
+		}
+	}
+
+	if isContainerPathFormat(destPath) {
+		name := strings.Split(destPath, ":")[0]
+		destination := strings.Split(destPath, ":")[1]
+		err := copyBetweenHostAndContainer(name, srcPath, destination, runtime, true)
+		if err != nil {
+			return errors.Wrapf(err, "error copy from %s to %s", srcPath, destPath)
+		}
+		return nil
+	}
+
+	if isContainerPathFormat(srcPath) {
+		name := strings.Split(srcPath, ":")[0]
+		srcPath = strings.Split(srcPath, ":")[1]
+		destination := destPath
+		err := copyBetweenHostAndContainer(name, srcPath, destination, runtime, false)
+		if err != nil {
+			return errors.Wrapf(err, "error copy from %s to %s", srcPath, destPath)
+		}
+		return nil
+	}
+	return errors.Errorf("invalid parameters: %s, %s", srcPath, destPath)
+}
+
+func copyBetweenHostAndContainer(name, srcPath, destination string, runtime *libpod.Runtime, isFromHostToContainer bool) error {
+	ctr, err := runtime.LookupContainer(name)
+	if err != nil {
+		return errors.Wrapf(err, "error looking up container %q", name)
+	}
+	mountPoint, err := ctr.Mount()
+	if err != nil {
+		return errors.Wrapf(err, "error mounting container %q", ctr.ID())
+	}
+	user, err := getUser(mountPoint, ctr.User())
+	if err != nil {
+		return err
+	}
+	idmappingopts, err := ctr.IDMappings()
+	if err != nil {
+		return errors.Wrapf(err, "error getting IDMappingOptions")
+	}
+	containerOwner := idtools.IDPair{UID: int(user.UID), GID: int(user.GID)}
+	hostUID, hostGID, err := getHostIDs(idmappingopts.UIDMap, idmappingopts.GIDMap, int(user.UID), int(user.GID))
+	if err != nil {
+		return err
+	}
+
+	hostOwner := idtools.IDPair{UID: int(hostUID), GID: int(hostGID)}
+
+	var dest string
+	if isFromHostToContainer {
+		dest = mountPoint
+		if destination != "" && filepath.IsAbs(destination) {
+			dest = filepath.Join(dest, destination)
+		} else {
+			if err = idtools.MkdirAllAndChownNew(filepath.Join(dest, ctr.WorkingDir()), 0755, hostOwner); err != nil {
+				return errors.Wrapf(err, "error creating directory %q", filepath.Join(dest, ctr.WorkingDir()))
+			}
+			dest = filepath.Join(dest, ctr.WorkingDir(), destination)
+		}
+	} else {
+		dest = destination
+		if destination == "" || !filepath.IsAbs(destination) {
+
+			dir, err := os.Getwd()
+			if err != nil {
+				return errors.Wrapf(err, "err getting current working directory")
+			}
+			if err = idtools.MkdirAllAndChownNew(filepath.Join(dir, dest), 0755, hostOwner); err != nil {
+				return errors.Wrapf(err, "error creating directory %q", filepath.Join(dir, dest))
+			}
+			dest = filepath.Join(dir, dest)
+		}
+
+	}
+
+	if len(destination) > 0 && destination[len(destination)-1] == os.PathSeparator {
```

Comment:
What's the os.PathSeparator check here? Isn't this the same as `filepath.IsAbs()`?

Reply:
Oh, is this a check to see if we're copying into the given directory, versus into a directory with a specific name?

Reply:
yes...from what I learned from buildah copy :thought_balloon: